clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/ReservationReportCachingAgent.groovy

```diff
@@ -31,51 +32,51 @@ import com.netflix.spinnaker.cats.agent.AgentDataType
 import com.netflix.spinnaker.cats.agent.CacheResult
 import com.netflix.spinnaker.cats.agent.CachingAgent
 import com.netflix.spinnaker.cats.agent.DefaultCacheResult
+import com.netflix.spinnaker.cats.cache.Cache
 import com.netflix.spinnaker.cats.cache.CacheData
 import com.netflix.spinnaker.cats.provider.ProviderCache
+import com.netflix.spinnaker.clouddriver.aws.data.Keys
 import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider
 import com.netflix.spinnaker.clouddriver.aws.security.AmazonCredentials
 import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials
 import com.netflix.spinnaker.clouddriver.aws.model.AmazonReservationReport
 import com.netflix.spinnaker.clouddriver.aws.provider.AwsProvider
+import com.netflix.spinnaker.clouddriver.cache.CustomScheduledAgent
 import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.ApplicationContext
 import rx.Observable
 import rx.Scheduler
 
 import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.TimeUnit
 import java.util.concurrent.atomic.AtomicInteger
 
 import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE
 import static com.netflix.spinnaker.clouddriver.aws.data.Keys.Namespace.RESERVATION_REPORTS
+import static com.netflix.spinnaker.clouddriver.aws.data.Keys.Namespace.RESERVED_INSTANCES
 
 @Slf4j
-class ReservationReportCachingAgent implements CachingAgent {
-  private static final Collection<AgentDataType> types = Collections.unmodifiableCollection([
+class ReservationReportCachingAgent implements CachingAgent, CustomScheduledAgent {
+  private static final long DEFAULT_POLL_INTERVAL_MILLIS = TimeUnit.MINUTES.toMillis(1)
+  private static final long DEFAULT_TIMEOUT_MILLIS = TimeUnit.MINUTES.toMillis(5)
+
+  final Collection<AgentDataType> types = Collections.unmodifiableCollection([
     AUTHORITATIVE.forType(RESERVATION_REPORTS.ns)
   ])
 
   private final Scheduler scheduler
+  private Cache cacheView
 
-  @Override
-  String getProviderName() {
-    AwsProvider.PROVIDER_NAME
-  }
-
-  @Override
-  String getAgentType() {
-    "${ReservationReportCachingAgent.simpleName}"
-  }
-
-  @Override
-  Collection<AgentDataType> getProvidedDataTypes() {
-    types
-  }
 
   final AmazonClientProvider amazonClientProvider
   final Collection<NetflixAmazonCredentials> accounts
   final ObjectMapper objectMapper
   final AccountReservationDetailSerializer accountReservationDetailSerializer
 
+  @Autowired
```

Comment:
why autowire this instead of just constructor injecting it like the rest of its dependencies? I'm pretty sure ApplicationContext is available where this thing gets instantiated